File: 645.set-mismatch/solution.py
from typing import List

class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        expectSum = (n * (n + 1)) >> 1
        s = 0
        results = [0, 0]
        memo = set()
        for i, num in enumerate(nums):
            s = s + num
            if (num in memo):
                results[0] = num
            memo.add(num)
        results[1] = (expectSum - s) + results[0]
        return results

File: 645.set-mismatch/solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        int[] results = new int[] {0, 0};
        int expectSum = (n * (n + 1)) >> 1;
        int sumOfNums = 0;
        
        for (int i = 0; i < n; i++) {
            int val = Math.abs(nums[i]);
            sumOfNums += val;
            
            if (nums[val - 1] < 0) {
                results[0] = val;
            }
            nums[val - 1] *= -1;
        }
        results[1] = results[0] + expectSum - sumOfNums;
        return results;
    }
}